exporters/dynamicconfig/confignotifier.go

```diff
@@ -145,9 +156,9 @@ func (n *ConfigNotifier) Stop() {
 
 func (n *ConfigNotifier) Register(watcher ConfigWatcher) {
 	n.lock.Lock()
-	defer n.lock.Unlock()
-
 	n.subscribed[watcher] = true
+	n.lock.Unlock()
+
 	watcher.OnInitialConfig(n.config)
```

Comment:
Make a copy of n.config in the critical section and use it outside the critical section.

Reply:
TODO: Do we have a go equivalent of C++'s thread-safety analysis?

https://clang.llvm.org/docs/ThreadSafetyAnalysis.html

Reply:
Go tsan.

Reply:
By go tsan, do you mean this? https://github.com/google/sanitizers

So open telemetry actually has tsan enabled by default in its makefile! `make test` also looks for race conditions